### ISSUE TITLE
Panic while type asserting FieldRows to map[string]interface{} in rawToOneAPIErr

### ISSUE THREAD

**Issue by 1lann** [Azure/azure-kusto-go#42]
I got the following panic while trying to query data. Let me know on Teams or email if you need more specific details (my alias is jasch). It doesn't seem to always happen as I'm unable to reproduce it consistently. The `kusto.ResultsProgressiveDisable()` is being used on the query that causes the panic.

```
panic: interface conversion: interface {} is []interface {}, not map[string]interface {}

goroutine 210 [running]:
github.com/Azure/azure-kusto-go/kusto/internal/frames/v2.rawToOneAPIErr(0xc01ee00000, 0x43bb7d4, 0x7fffe00, 0x7d0000, 0xc0152cb4d0, 0x8b5320)
		/home/jason/.go/pkg/mod/github.com/!azure/azure-kusto-go@v0.3.1/kusto/internal/frames/v2/v2.go:217 +0x17f
github.com/Azure/azure-kusto-go/kusto/internal/frames/v2.(*DataTable).UnmarshalRaw(0xc0152cb4d0, 0xc01ee00000, 0x43bb7d4, 0x7fffe00, 0x0, 0x0)
		/home/jason/.go/pkg/mod/github.com/!azure/azure-kusto-go@v0.3.1/kusto/internal/frames/v2/v2.go:61 +0x14f
github.com/Azure/azure-kusto-go/kusto/internal/frames/v2.(*Decoder).decode(0xc0000fa040, 0x8bb868, 0xc009e57980, 0xc0129fe000, 0x0, 0x0)
		/home/jason/.go/pkg/mod/github.com/!azure/azure-kusto-go@v0.3.1/kusto/internal/frames/v2/decoder.go:117 +0xb1e
github.com/Azure/azure-kusto-go/kusto/internal/frames/v2.(*Decoder).decodeFrames(0xc0000fa040, 0x8bb868, 0xc009e57980, 0xc0129fe000)
		/home/jason/.go/pkg/mod/github.com/!azure/azure-kusto-go@v0.3.1/kusto/internal/frames/v2/decoder.go:83 +0x85
github.com/Azure/azure-kusto-go/kusto/internal/frames/v2.(*Decoder).Decode.func1(0x8b7ca0, 0xc0000fe840, 0xc0129fe000, 0xc0000fa040, 0x8bb868, 0xc009e57980)
		/home/jason/.go/pkg/mod/github.com/!azure/azure-kusto-go@v0.3.1/kusto/internal/frames/v2/decoder.go:60 +0x3cc
created by github.com/Azure/azure-kusto-go/kusto/internal/frames/v2.(*Decoder).Decode
		/home/jason/.go/pkg/mod/github.com/!azure/azure-kusto-go@v0.3.1/kusto/internal/frames/v2/decoder.go:33 +0x147
```

**Comment by element-of-surprise** [comment]
This is a "my bad". Here is what's going on:

Your stream is receiving one of two things:
- Data that doesn't conform to the expected JSON message
-  An error that was supposed to be an old oneAPI error
 
**This is why you aren't tripping on it all the time, your error is transient.**  

The code attempts to convert what it thinks is an error to OneAPI error type.  These types of errors are a recursive error type (it can contain itself).  The code makes a bad assumption when it tries to unpack one or more recursive entries:

```go
if err := errors.OneToErr(oe.(map[string]interface{}), op); err != nil {
 ...
```
I shouldn't have made any assumptions about the type.  One of the only things in the API I don't like is the OneAPI stuff.  Sooner or later I'm going to strip it and just return the raw response.  It isn't providing value.  

This isn't related to this particular issue, but I wouldn't use kusto.ResultsProgressiveDisable().  The thing is, progressive mode in Go, at least at the time when I wrote this, could decode faster than the server can send. I would test Progressive against .Net's non-progressive mode and it was the exact same speed. This is a good indication that the backends were the bottleneck. The overhead of progressive mode is trivial.

Go's non-progressive mode is not-efficient in comparison to progressive, as it doesn't stream, it reads the entire contents into memory and then returns the result.  You could be processing your results as they come back instead of waiting unless your use case requires all data to do anything.

When I wrote this, several of us did not understand non-progressive mode well enough.  Because of how the implementers chose to send the message, it looks like one big JSON message to read into memory.  In reality, it is abusing JSON and should be interpreted as a set of datagrams representing a streaming message with the single JSON wrappers as the beginning and end of the data.

At some point I plan to fix that, but it hasn't seemed very important as the speed difference between the two is nil.

**Comment by AsafMah** [comment]
Solved in 0.6.0, feel free to re-open if you have issues.
